libkbfs/tlf_journal.go

```diff
@@ -809,6 +808,62 @@ func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
 	}, nil
 }
 
+func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	return j.getBaseJournalStatusLocked()
+}
+
+func (j *tlfJournal) getJournalStatusWithPaths(ctx context.Context,
+	config Config, mdOps journalMDOps, blocks *folderBlockOps) (
+	TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	status, err := j.getBaseJournalStatusLocked()
+	if err != nil {
+		return TLFJournalStatus{}, err
+	}
+
+	// Make chains over the entire range to get the unflushed files.
+	// TODO: cache this chains object and update it when new revisions
+	// appear and when revisions are flushed, instead of rebuilding
+	// every time.
+	if status.RevisionEnd != MetadataRevisionUninitialized {
+		var irmds []ImmutableRootMetadata
+		// These GetRange calls end up taking journalLock for reading
```

Comment:
I disagree about consistent results -- I think it's pretty important to have them.  Who knows what could end up relying on .kbfs_status down the line.

But you're right about being able to get full results from `mdOps` regardless of underlying journal flushes.

I'm most worried about the case where locally we switch to a branch, and then `mdOps` returns revisions that were written by some other device when we ask for `Merged` revisions in the range -- the range could even appear complete.  I suppose I can check the device key of all the revisions to be certain . . .